### PR TITLE
Fix tvOS build

### PR DESCRIPTION
**Pull request by douglowder**
Changes to fix compile errors on tvOS:

- Add `TARGET_OS_TV` to Apple platforms in `folly-config.h.cmake`
- Exclude TV from adding duplicate time and clock definitions in `Time.h`

Test strategy:

- Build and test RNTester for `react-native-tvos` fork 0.73 branch with the above Folly changes
- Apple TV should compile and run

### CODE REVIEW

File: folly/portability/Time.h [hunk 1]
```diff
@@ -40,7 +40,7 @@
 
 // These aren't generic implementations, so we can only declare them on
 // platforms we support.
-#if !FOLLY_HAVE_CLOCK_GETTIME && (defined(__MACH__) || defined(_WIN32))
+#if !FOLLY_HAVE_CLOCK_GETTIME && (defined(__MACH__) || defined(_WIN32)) && !(defined(TARGET_OS_TV) && TARGET_OS_TV)
```
Comment (NickGerleman) [inline folly/portability/Time.h:43]: FWIW this is broken on iOS with XCode as well. Needed to ad `FOLLY_HAVE_CLOCK_GETTIME` explicitly in RN ☹️.

Comment (douglowder) [reply folly/portability/Time.h:43]: Could we not add this in `folly-config.h.cmake`?

Comment (NickGerleman) [reply folly/portability/Time.h:43]: Which part? The iOS RN build doesn’t use CMake.

Comment (douglowder) [reply folly/portability/Time.h:43]: Aha now I see the source of my confusion... I was testing Folly from main with RN from 0.73, the Podfile for that branch is different. I'll test with that, and make changes to this PR. Thanks!!!

Comment (douglowder) [reply folly/portability/Time.h:43]: @NickGerleman thanks for the help with this -- I tested this today, and if the RCT-Folly podspec has `FOLLY_HAVE_CLOCK_GETTIME=1` added, this patch is not needed. Closing the PR.

**Comment by douglowder** [comment]
Not needed (see inline comments). Closing.

**Pull request closed without merging**